distribution/metadata/v2_metadata_service.go

```diff
@@ -15,10 +18,24 @@ type V2MetadataService struct {
 
 // V2Metadata contains the digest and source repository information for a layer.
 type V2Metadata struct {
+	AuthConfigHash   string
 	Digest           digest.Digest
 	SourceRepository string
 }
 
+// HashAuthConfig returns an sha256 sum of the "authConfig" if successful.
+func HashAuthConfig(authConfig *types.AuthConfig) (string, error) {
+	if authConfig == nil {
+		return "", nil
+	}
+	buf, err := json.Marshal(authConfig)
```

Comment:
This was partially why I suggested just using the username. Another is that updates to the auth config structure could cause the hash to change.